src/dev/code_coverage/ingest_coverage/transforms.js

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { left, right } from './either';
+import { always, id, noop } from './utils';
+
+const maybeTotal = x => (x === 'total' ? left(x) : right(x));
+
+export const trimLeftFrom = (text, x) => x.replace(new RegExp(`(?:.*)(${text}.*$)`, 'gm'), '$1');
+
+export const statsAndstaticSiteUrl = (...xs) => {
+  const [staticSiteUrl] = xs[0][1];
+  const [stats] = xs[0];
+  return {
+    staticSiteUrl,
+    ...stats,
+  };
+};
+
+export const addJsonSummaryPath = jsonSummaryPath => obj => ({
+  jsonSummaryPath: trimLeftFrom('target', jsonSummaryPath),
+  ...obj,
+});
+
+export const truncate = text => obj => {
+  const { staticSiteUrl } = obj;
+  if (staticSiteUrl.includes(text)) obj.staticSiteUrl = trimLeftFrom(text, staticSiteUrl);
+  return obj;
+};
+
+export const addTimeStamp = ts => obj => ({
+  ...obj,
+  '@timestamp': ts,
+});
+
+const splitOnRoot = root => x => x.split(root)[1];
```

Comment:
Why the nested function? Why not just have a function which takes two arguments? It also doesn't split, which is misleading, maybe `stripRoot`? It could also be simplified:

`const stripRoot = (root, fullPath) => fullPath.replace(root, '')`

Reply:
Sure!

Reply:
It splits it into an array and grabs the second element, essentially stripping the root from the string.

```
const splitOnRoot = root => x => x.split(root)[1];
splitOnRoot('/foo/bar')('/foo/bar/baz/quux')
> "/baz/quux"
```

```
const stripRoot = (root, fullPath) => fullPath.replace(root, '')
stripRoot('/foo/bar', '/foo/bar/baz/quux');
> "/baz/quux"
```

Reply:
If that is correct, I think using a replace is much better than first converting to an array.

Reply:
Sure!